src/Compiler/Microsoft.AspNetCore.Razor.Language/src/Components/ComponentDesignTimeNodeWriter.cs

```diff
@@ -596,6 +623,102 @@ public override void WriteComponentAttribute(CodeRenderingContext context, Compo
+
+        context.CodeWriter.Write(DesignTimeVariable);
+        context.CodeWriter.Write(" = ");
+
+        if (componentNode.TypeInferenceNode == null)
+        {
+            context.CodeWriter.Write("((global::");
+            context.CodeWriter.Write(componentNode.Component.GetTypeNamespace());
+            context.CodeWriter.Write(".");
+            context.CodeWriter.Write(componentNode.Component.GetTypeNameIdentifier());
+            if (componentNode.Component.IsGenericTypedComponent())
+            {
+                // If there are generic type components, but no type inference node, then it means
+                // the user specified the type parameters, so we can use them directly
+                context.CodeWriter.Write("<");
+
+                var i = 0;
+                foreach (var typeArgumentNode in componentNode.Children.OfType<ComponentTypeArgumentIntermediateNode>())
+                {
+                    if (i++ > 0)
+                    {
+                        context.CodeWriter.Write(", ");
+                    }
+
+                    foreach (var token in typeArgumentNode.Children.OfType<IntermediateToken>())
+                    {
+                        // As per WriteComponentTypeArgument, we expect every token to be C#, but check just in case
+                        if (token.IsCSharp)
+                        {
+                            context.CodeWriter.Write(token.Content);
+                        }
+                        else
+                        {
+                            Debug.Fail($"Unexpected non-C# content in a generic type parameter: '{token.Content}'");
+                        }
+                    }
+                }
+                context.CodeWriter.Write(">");
+            }
+            context.CodeWriter.Write(")default)");
+        }
+        else
+        {
+            if (typeInferenceLocalName is null)
+            {
+                throw new InvalidOperationException("No type inference local name was supplied, but type inference is required to reference a component type.");
+            }
+
+            // Earlier when we did the type inference stuff, we captured a variable which the compiler would know the type information
+            // for explicitly for the purposes of using it now
+            context.CodeWriter.Write(typeInferenceLocalName);
+        }
+
+        context.CodeWriter.Write(".");
+        context.CodeWriter.WriteLine();
+
+        using (context.CodeWriter.BuildLinePragma(attributeSourceSpan, context))
+        {
+            context.CodeWriter.WritePadding(0, attributeSourceSpan, context);
+            context.AddSourceMappingFor(attributeSourceSpan);
+            context.CodeWriter.WriteLine(node.PropertyName);
+        }
+
+        // When doing a Find All Refs, sometimes the text content of the generated file shows up,
+        // so being able to see "= default" in there would confuse users, so keep it on a separate line
```

Comment:
What's this comment talking about? Where's the `= default`?

Reply:
Good catch, thanks! Missed updating this when I changed the code gen after one of the above comments